.github/workflows/build_and_test.yml

```diff
@@ -74,11 +74,54 @@ jobs:
         shell: bash
         id: build-tag
 
-  check-codestyle-python:
+      - name: Get buildtools tag
+        run: |
+          if [[ "$GITHUB_EVENT_NAME" == "pull_request" ]]; then
+            IMAGE_TAG=$GITHUB_RUN_ID
+          else
+            IMAGE_TAG=pinned
+          fi
+          
+          echo "image_tag=${IMAGE_TAG}" >> $GITHUB_OUTPUT
+        shell: bash
+        id: buildtools-tag
+
+  check-if-build-tools-dockerfile-changed:
     needs: [ check-permissions ]
+    runs-on: ubuntu-latest
+    outputs:
+      docker_file_changed: ${{ steps.dockerfile.outputs.docker_file_changed }}
+    steps:
+      - name: Check if Dockerfile.buildtools has changed
+        id: dockerfile
+        run: |
+          if [[ ${{ github.event.pull_request.id }} == "" ]]; then
```

Comment:
GitHub templates (`${{ ... }}`) are evaluated before the run, so the result bash is going to have incorrect syntax

```bash
if [[  == "" ]]; then
```

You can use `"${GITHUB_EVENT_NAME}" == "pull_request"`, as we already do in "Get buildtools tag" step.